File: astrasphere-quantum-research-os/astrasphere-quantum-research-os/backend/app/models/research.py
"""ORM models for research projects."""

import uuid
from datetime import datetime

from sqlalchemy import DateTime, String, Text, func
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import Mapped, mapped_column

from app.db.base import Base


class ResearchProject(Base):
    """A quantum research project tracked in AstraSphere."""

    __tablename__ = "research_projects"

    id: Mapped[uuid.UUID] = mapped_column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    title: Mapped[str] = mapped_column(String(255), nullable=False, index=True)
    summary: Mapped[str] = mapped_column(Text, nullable=False, default="")
    status: Mapped[str] = mapped_column(String(50), nullable=False, default="draft")
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), server_default=func.now(), nullable=False
    )
    updated_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True),
        server_default=func.now(),
        onupdate=func.now(),
        nullable=False,
    )

File: astrasphere-quantum-research-os/astrasphere-quantum-research-os/frontend/src/types/research.ts
export type ResearchStatus = 'draft' | 'active' | 'completed' | 'archived';

export interface ResearchProject {
  id: string;
  title: string;
  summary: string;
  status: ResearchStatus;
  created_at: string;
  updated_at: string;
}

export interface ResearchProjectCreateInput {
  title: string;
  summary?: string;
  status?: ResearchStatus;
}
